lib/engine/gameutils/undo_helper.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module Engine
+  module GameUtils
+    class UndoHelper
+      attr_reader :undo_list
+
+      def initialize(actions = nil)
+        process_actions(actions)
+      end
+
+      def process_actions(actions)
+        @undo_list = []
+        @last_known_undo = 0
+        return if actions.nil?
```

Comment:
in generally i like do to return unless actions instead 

but i don't think we need this pattern anayws if actions is just an empty list